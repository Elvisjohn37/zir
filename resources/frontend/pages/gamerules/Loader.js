import React from 'react';
import styles from './Loader.module';
import Grid from '@mui/material/Grid';
import Skeleton from 'frontend/components/Skeleton';
import lodash from 'lodash';

export default function Loader({ className }) {
	return (
		<Grid container spacing={1} className={className}>
			{lodash.range(9).map((item, index) => (
				<Grid item md={6} xs={12} lg={4} key={index}>
					<Skeleton count={1} height={36} className={styles.content} />
				</Grid>
			))}
		</Grid>
	);
}
