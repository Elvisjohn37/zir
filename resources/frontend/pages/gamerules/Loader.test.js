import React from 'react';
import { render } from '@testing-library/react';
import Loader from '../gamerules/Loader';

jest.mock('@mui/material/Grid');
jest.mock('frontend/components/Skeleton');
jest.mock('lodash');

describe('<Loader />', () => {
	it('should match snapshot', () => {
		let { asFragment } = render(<Loader />);

		expect(asFragment()).toMatchSnapshot();
	});
});
