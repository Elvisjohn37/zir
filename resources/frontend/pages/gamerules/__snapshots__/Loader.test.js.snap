// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    container="true"
    data-testid="Grid"
    spacing="1"
  >
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
    <div
      data-testid="Grid"
      item="true"
      lg="4"
      md="6"
      xs="12"
    >
      <div
        class="content"
        count="1"
        data-testid="Skeleton"
        height="36"
      />
    </div>
  </div>
</DocumentFragment>
`;
